plugins/notifiers/slack/client.go

```diff
@@ -33,27 +34,38 @@ type userResponse struct {
 	Error string `json:"error"`
 }
 
+type SlackWorkspace struct {
+	WorkspaceName string `mapstructure:"workspace" validate:"required"`
+	AccessToken   string `mapstructure:"access_token" validate:"required"`
+	Criteria      string `mapstructure:"criteria" validate:"required"`
+}
+
 type Notifier struct {
-	accessToken string
+	workspaces []SlackWorkspace
 
-	slackIDCache        map[string]string
+	slackIDCache        map[string]*slackIDCacheItem
 	Messages            domain.NotificationMessages
 	httpClient          utils.HTTPClient
 	defaultMessageFiles embed.FS
 }
 
+type slackIDCacheItem struct {
+	SlackID   string
+	Workspace SlackWorkspace
```

Comment:
given that this struct is the cached value, this `SlackWorkspace` is going to be referenced as many as the cached items, should we change it to pointer type `Workspace *SlackWorkspace`?

Reply:
updated